trajopt/include/trajopt/file_write_callback.hpp

```diff
@@ -0,0 +1,9 @@
+#pragma once
+#include <trajopt/problem_description.hpp>
+#include <trajopt_sco/optimizers.hpp>
+
+namespace trajopt
+{
+Optimizer::Callback WriteCallback(std::shared_ptr<std::ofstream> file,
+                                  TrajOptProbPtr prob);
```

Comment:
```suggestion
                                  const TrajOptProb& prob);
```
Also probably want to update the plot callback to match.